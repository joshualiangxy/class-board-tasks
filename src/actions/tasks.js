import firebase from '../firebase/firebase';

export const addPersonalTask = task => ({ type: 'ADD_PERSONAL_TASK', task });

export const startAddPersonalTask = task => {
  return (dispatch, getState) => {
    const uid = getState().auth.user.uid;
    const id = task.id;

    return firebase
      .firestore()
      .collection('users')
      .doc(uid)
      .collection('tasks')
      .doc(id)
      .set(task)
      .then(() => dispatch(addPersonalTask(task)));
  };
};

export const removePersonalTask = id => ({ type: 'REMOVE_PERSONAL_TASK', id });

export const startRemovePersonalTask = id => {
  return (dispatch, getState) => {
    const uid = getState().auth.user.uid;

    return firebase
      .firestore()
      .collection('users')
      .doc(uid)
      .collection('tasks')
      .doc(id)
      .delete()
      .then(() => dispatch(removePersonalTask(id)));
  };
};

export const editPersonalTask = (id, updates) => ({
  type: 'EDIT_PERSONAL_TASK',
  id,
  updates
});

export const startEditPersonalTask = (id, updates) => {
  return (dispatch, getState) => {
    const uid = getState().auth.user.uid;

    return firebase
      .firestore()
      .collection('users')
      .doc(uid)
      .collection('tasks')
      .doc(id)
      .set(updates)
      .then(dispatch(editPersonalTask(id, updates)));
  };
};

export const setTasks = tasks => ({ type: 'SET_TASK', tasks });

export const startSetTasks = () => {
  return (dispatch, getState) => {
    const uid = getState().auth.user.uid;
    const tasks = [];
    const userRef = firebase.firestore().collection('users').doc(uid);
    const tasksRef = userRef.collection('tasks');

    return tasksRef
      .get()
      .then(querySnapshot =>
        querySnapshot.forEach(queryDocSnapshot =>
          tasks.push({ id: queryDocSnapshot.id, ...queryDocSnapshot.data() })
        )
      )
      .then(() => {
        userRef.get().then(userDocSnapshot => {
          const groupIds = userDocSnapshot.get('groups') || [];
          const groupsPromises = [];

          groupIds.forEach(gid => {
            const groupPromises = [];
            const groupRef = firebase.firestore().collection('groups').doc(gid);

            groupPromises.push(
              groupRef
                .get()
                .then(groupDocSnapshot => groupDocSnapshot.get('name'))
            );
            groupPromises.push(groupRef.collection('tasks').get());

            groupsPromises.push(
              Promise.all(groupPromises).then(
                ([groupName, queryTaskCollectionSnapshot]) =>
                  queryTaskCollectionSnapshot.forEach(queryTaskDocSnapshot =>
                    tasks.push({
                      groupName,
                      gid,
                      id: queryTaskDocSnapshot.id,
                      ...queryTaskDocSnapshot.data()
                    })
                  )
              )
            );

            return Promise.all(groupPromises);
          });
        });
      })
      .then(() => dispatch(setTasks(tasks)));
  };
};

export const toggleCompletedPersonal = (id, completedState) => ({
  type: 'TOGGLE_COMPLETED_PERSONAL',
  id,
  completedState
});

export const startToggleCompletedPersonal = (id, completedState) => {
  return (dispatch, getState) => {
    const uid = getState().auth.user.uid;

    return firebase
      .firestore()
      .collection('users')
      .doc(uid)
      .collection('tasks')
      .doc(id)
      .update({ completed: !completedState })
      .then(() => toggleCompletedPersonal(id, completedState));
  };
};

export const removeTaskData = () => ({ type: 'REMOVE_TASK_DATA' });
