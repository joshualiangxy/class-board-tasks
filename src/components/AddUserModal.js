import React, { useState } from 'react';
import Modal from 'react-modal';
import firebase from '../firebase/firebase';

// Note: This still works off uid for now. This also does NOT work yet.
const AddUserModal = ({ isOpen, onRequestClose, group }) => {
  const addNewUser = (userEmail, group) => {
    const uid = userEmail; // TODO: Change this when email-uid are linked!

    // Both of these references are DocumentReference
    const userRef = firebase.firestore().collection('users').doc(uid);
    const groupRef = firebase.firestore().collection('groups').doc(group);

    // Gets the data from a user.
    userRef.get().then(doc => {
      const data = doc.data();
      // Write displayName, studentNum, and UID into a new document
      if (typeof data === 'undefined') {
        // Most likely a user that doesn't exist.
        setError('No such user found');
      } else {
        groupRef
          .collection('users')
          .doc(uid) // Create document for this user's UID
          .set({
            studentNum: data.studentNum,
            displayName: data.displayName,
            uid: uid,
            admin: false
          })
          .then(() => {
            setError('User added!');
          })
          .catch(error => {
            setError('Something went wrong.');
          });
      }
      // Defaults to non-admin.
    });
  };

  // TODO: When calling this modal, pass in the group UID as a prop
  const [userEmail, setUserEmail] = useState('');
  const [error, setError] = useState('');
  // const [usersAdded, setUsersAdded] = useState([]); // For listing down people that have been added?

  const onUserEmailChange = e => {
    setUserEmail(e.target.value);
  };

  const onCancel = () => {
    setUserEmail('');
    onRequestClose();
  };

  const onSubmit = e => {
    e.preventDefault();
    const submittedEmail = userEmail.trim();
    if (!submittedEmail) {
      setError('Please enter a group name');
    } else {
      addNewUser(submittedEmail, group);
      // TODO: Catch user not found
      // Does not close after submitting, so that the user can continue to add more users

      // setUserEmail(''); // Also resets the field.
      // setError('Added!');
    }
  };

  return (
    <Modal
      isOpen={isOpen}
      contentLabel="New Group"
      onRequestClose={() => onRequestClose()}
      appElement={document.getElementById('root')}
    >
      <form onSubmit={onSubmit}>
        <input
          type="text"
          placeholder="User ID (required)"
          value={userEmail}
          onChange={onUserEmailChange}
          autoFocus
        />
        <button>Submit</button>
        <button onClick={onCancel}>Cancel</button>
        <div>{error}</div>
      </form>
    </Modal>
  );
};

export default AddUserModal;
