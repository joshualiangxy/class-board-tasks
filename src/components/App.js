import React from 'react';
import '../styles/App.css';

const App = () => (
  <div>
    <h1>This is my app</h1>
  </div>
);

export default App;

