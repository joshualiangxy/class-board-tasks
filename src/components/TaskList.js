import React from 'react';
import { connect } from 'react-redux';
import getSortedTasks from '../selectors/tasks';
import TaskListHeader from './TaskListHeader';
import TaskListGroup from './TaskListGroup';
import TaskListItem from './TaskListItem';

export const TaskList = ({ sortedList, grouped }) => (
  <div className="content-container list">
    <TaskListHeader />
    {Object.keys(sortedList).length === 1 &&
    sortedList.others.tasks.length === 0 ? (
      <div className="list-item__header list-item__header-message">
        <p>No tasks yet</p>
      </div>
    ) : grouped ? (
      Object.values(sortedList).map((group, id) => {
        if (group.tasks.length > 0)
          return (
            <TaskListGroup key={Object.keys(sortedList)[id]} group={group} />
          );
      })
    ) : (
      Object.values(sortedList)[0].tasks.map(task => (
        <TaskListItem
          key={task.id}
          task={task}
          showGroup={true}
          dashboard={true}
        />
      ))
    )}
  </div>
);

const mapStateToProps = ({ tasks, filters }) => ({
  sortedList: getSortedTasks(tasks, filters),
  grouped: filters.grouped
});

export default connect(mapStateToProps)(TaskList);
