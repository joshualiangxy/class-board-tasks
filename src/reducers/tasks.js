const tasksReducer = (state = [], action) => {
  switch (action.type) {
    case 'ADD_TASK':
      return [...state, action.task];
    case 'REMOVE_TASK':
      return state.filter(task => task.id !== action.id);
    case 'EDIT_TASK':
      return state.map(task =>
        task.id === action.id
          ? {
              ...task,
              ...action.updates
            }
          : task
      );
    case 'SET_TASK':
      return action.tasks;
    default:
      return state;
  }
};

export default tasksReducer;
