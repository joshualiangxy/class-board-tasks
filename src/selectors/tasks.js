const getSortedTasks = (tasks, { text, sortBy, grouped }) => {
  const lowercaseText = text.toLowerCase();

  const processedTasks = tasks
    .filter(
      task =>
        task.title.toLowerCase().includes(lowercaseText) ||
        task.description.toLowerCase().includes(lowercaseText) ||
        task.module.toLowerCase().includes(lowercaseText)
    )
    .sort((taskOne, taskTwo) => {
      switch (sortBy) {
        case 'deadline':
          if (taskOne.deadline && taskTwo.deadline)
            return taskOne.deadline - taskTwo.deadline;
          else if (taskOne.deadline) return -1;
          else if (taskTwo.deadline) return 1;
          else return taskOne.title.localeCompare(taskTwo.title);
        case 'name':
          return taskOne.title.localeCompare(taskTwo.title);
      }
    });

  const sortedList = {};

  if (grouped) {
    processedTasks.forEach(task => {
      if (task.groupName) {
        const gid = task.gid;

        if (!sortedList[gid])
          sortedList[gid] = { groupName: task.groupName, tasks: [] };
        sortedList[gid].tasks.push(task);
      } else if (task.module) {
        const module = task.module;

        if (!sortedList[module])
          sortedList[module] = { groupName: task.module, tasks: [] };
        sortedList[module].tasks.push(task);
      } else {
        if (!sortedList.others)
          sortedList.others = { groupName: 'others', tasks: [] };
        sortedList.others.tasks.push(task);
      }
    });
  } else sortedList.others = { groupName: 'others', tasks: processedTasks };

  return sortedList;
};

export default getSortedTasks;
