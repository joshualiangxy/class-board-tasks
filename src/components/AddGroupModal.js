import React, { useState } from 'react';
import Modal from 'react-modal';
import { startNewGroup } from '../actions/groups';
import { connect } from 'react-redux';

// TODO: Add the ability to add other users into the group
const AddGroupModal = ({ isOpen, onRequestClose, startNewGroup }) => {
  const [groupName, setGroupName] = useState('');
  const [module, setModule] = useState('');
  const [error, setError] = useState('');

  const onGroupNameChange = e => setGroupName(e.target.value);

  const onModuleChange = e => {
    const module = e.target.value.toUpperCase();

    setModule(module);
  };

  const onCancel = () => {
    setError('');
    setGroupName('');
    setModule('');
    onRequestClose();
  };

  const onSubmit = e => {
    e.preventDefault();
    const submittedName = groupName.trim();
    const submittedMod = module.trim();
    if (!submittedName) {
      setError('Please enter a group name');
    } else {
      setError('');
      setGroupName('');
      setModule('');
      startNewGroup(submittedName, submittedMod).then(() => onRequestClose());
    }
  };

  return (
    <Modal
      isOpen={isOpen}
      contentLabel="New Group"
      onRequestClose={onCancel}
      closeTimeoutMS={200}
      appElement={document.getElementById('root')}
    >
      <form onSubmit={onSubmit}>
        <input
          type="text"
          placeholder="Group name (required)"
          value={groupName}
          onChange={onGroupNameChange}
          autoFocus
        />
        <input
          type="text"
          placeholder="Module code/Class name (Optional)"
          value={module}
          onChange={onModuleChange}
        />
        <button>Submit</button>
        <button onClick={onCancel}>Cancel</button>
        <div>{error}</div>
      </form>
    </Modal>
  );
};

const mapDispatchToProps = dispatch => ({
  startNewGroup: (groupName, module) =>
    dispatch(startNewGroup(groupName, module))
});

export default connect(null, mapDispatchToProps)(AddGroupModal);
