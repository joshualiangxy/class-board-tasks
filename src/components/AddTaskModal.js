import React from 'react';
import Modal from 'react-modal';
import { connect } from 'react-redux';
import { startAddPersonalTask, startAddGroupTask } from '../actions/tasks';
import TaskForm from './TaskForm';

export const AddTaskModal = ({
  startAddPersonalTask,
  startAddGroupTask,
  removeAdmin,
  isOpen,
  onRequestClose,
  gid,
  groupModule,
  groupName,
  addGroupTask,
  users
}) => {
  const submitTask = task =>
    gid
      ? startAddGroupTask(task, groupName)
          .then(isAdmin => {
            if (isAdmin) return addGroupTask(task);
            else return removeAdmin();
          })
          .then(() => onRequestClose())
      : startAddPersonalTask(task).then(() => onRequestClose());

  return (
    <Modal
      className="task-modal"
      isOpen={isOpen}
      contentLabel="Add Task"
      onRequestClose={onRequestClose}
      closeTimeoutMS={200}
      appElement={document.getElementById('root')}
    >
      <h1>Add Task</h1>
      <TaskForm
        gid={gid}
        groupModule={groupModule}
        users={users}
        submitTask={submitTask}
        onRequestClose={onRequestClose}
      />
    </Modal>
  );
};

const mapDispatchToProps = dispatch => ({
  startAddPersonalTask: task => dispatch(startAddPersonalTask(task)),
  startAddGroupTask: (task, groupName) =>
    dispatch(startAddGroupTask(task, groupName))
});

export default connect(undefined, mapDispatchToProps)(AddTaskModal);
