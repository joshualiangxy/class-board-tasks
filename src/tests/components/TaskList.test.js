import React from 'react';
import { shallow } from 'enzyme';
import { TaskList } from '../../components/TaskList';
import getSortedTasks from '../../selectors/tasks';
import tasks from '../fixtures/tasks';
import {
  filtersGrouped,
  filtersDeadline,
  filtersName,
  filtersDeadlineReversed,
  filtersNameReversed
} from '../fixtures/filters';

let wrapper;

beforeEach(() => {
  wrapper = shallow(
    <TaskList
      sortedList={getSortedTasks(tasks, filtersGrouped)}
      grouped={false}
    />
  );
});

describe('render', () => {
  it('should render TaskList with no tasks text', () => {
    wrapper = shallow(
      <TaskList
        sortedList={getSortedTasks([], filtersGrouped)}
        grouped={true}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render TaskList with tasks in groups', () => {
    wrapper.setProps({ grouped: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render TaskList with tasks ungrouped', () => {
    wrapper.setProps({
      sortedList: getSortedTasks(tasks, filtersDeadline)
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should render TaskList with tasks sorted by name', () => {
    wrapper.setProps({
      sortedList: getSortedTasks(tasks, filtersName)
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should render TaskList with tasks sorted by deadline in reverse', () => {
    wrapper.setProps({
      sortedList: getSortedTasks(tasks, filtersDeadlineReversed)
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should render TaskList with tasks sorted by name in reverse', () => {
    wrapper.setProps({
      sortedList: getSortedTasks(tasks, filtersNameReversed)
    });

    expect(wrapper).toMatchSnapshot();
  });
});
