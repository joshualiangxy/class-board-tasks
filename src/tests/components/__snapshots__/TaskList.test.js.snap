// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render TaskList with no tasks text 1`] = `
<div
  className="content-container list"
>
  <Connect(TaskListHeader) />
  <div
    className="list-item__header list-item__header-message"
  >
    <p>
      No tasks yet
    </p>
  </div>
</div>
`;

exports[`render should render TaskList with tasks in groups 1`] = `
<div
  className="content-container list"
>
  <Connect(TaskListHeader) />
  <TaskListGroup
    group={
      Object {
        "groupName": "CS2030",
        "tasks": Array [
          Object {
            "completed": false,
            "deadline": 1591329600000,
            "description": "Catch up on lectures!",
            "id": "5feb18c0-78a8-49b6-8584-75b8c1b40abc",
            "module": "CS2030",
            "title": "Watch lecture",
          },
          Object {
            "completed": false,
            "description": "",
            "id": "a1050220-fa88-4dc2-aa9d-9521ae95de47",
            "module": "CS2030",
            "title": "Do lab",
          },
        ],
      }
    }
    key="CS2030"
  />
  <TaskListGroup
    group={
      Object {
        "groupName": "others",
        "tasks": Array [
          Object {
            "completed": true,
            "description": "",
            "id": "d2b10492-853d-4290-baf8-cb47cd4ec04e",
            "module": "",
            "title": "Read a book everyday",
          },
        ],
      }
    }
    key="others"
  />
  <TaskListGroup
    group={
      Object {
        "groupName": "CS2040S",
        "tasks": Array [
          Object {
            "completed": true,
            "description": "",
            "id": "73c8ac1a-1d06-46ee-a39e-e656e1e41db0",
            "module": "CS2040S",
            "title": "Do problem set",
          },
        ],
      }
    }
    key="CS2040S"
  />
</div>
`;

exports[`render should render TaskList with tasks sorted by deadline in reverse 1`] = `
<div
  className="content-container list"
>
  <Connect(TaskListHeader) />
  <Connect(TaskListItem)
    dashboard={true}
    key="a1050220-fa88-4dc2-aa9d-9521ae95de47"
    showGroup={true}
    task={
      Object {
        "completed": false,
        "description": "",
        "id": "a1050220-fa88-4dc2-aa9d-9521ae95de47",
        "module": "CS2030",
        "title": "Do lab",
      }
    }
  />
  <Connect(TaskListItem)
    dashboard={true}
    key="5feb18c0-78a8-49b6-8584-75b8c1b40abc"
    showGroup={true}
    task={
      Object {
        "completed": false,
        "deadline": 1591329600000,
        "description": "Catch up on lectures!",
        "id": "5feb18c0-78a8-49b6-8584-75b8c1b40abc",
        "module": "CS2030",
        "title": "Watch lecture",
      }
    }
  />
  <Connect(TaskListItem)
    dashboard={true}
    key="d2b10492-853d-4290-baf8-cb47cd4ec04e"
    showGroup={true}
    task={
      Object {
        "completed": true,
        "description": "",
        "id": "d2b10492-853d-4290-baf8-cb47cd4ec04e",
        "module": "",
        "title": "Read a book everyday",
      }
    }
  />
  <Connect(TaskListItem)
    dashboard={true}
    key="73c8ac1a-1d06-46ee-a39e-e656e1e41db0"
    showGroup={true}
    task={
      Object {
        "completed": true,
        "description": "",
        "id": "73c8ac1a-1d06-46ee-a39e-e656e1e41db0",
        "module": "CS2040S",
        "title": "Do problem set",
      }
    }
  />
</div>
`;

exports[`render should render TaskList with tasks sorted by name 1`] = `
<div
  className="content-container list"
>
  <Connect(TaskListHeader) />
  <Connect(TaskListItem)
    dashboard={true}
    key="a1050220-fa88-4dc2-aa9d-9521ae95de47"
    showGroup={true}
    task={
      Object {
        "completed": false,
        "description": "",
        "id": "a1050220-fa88-4dc2-aa9d-9521ae95de47",
        "module": "CS2030",
        "title": "Do lab",
      }
    }
  />
  <Connect(TaskListItem)
    dashboard={true}
    key="5feb18c0-78a8-49b6-8584-75b8c1b40abc"
    showGroup={true}
    task={
      Object {
        "completed": false,
        "deadline": 1591329600000,
        "description": "Catch up on lectures!",
        "id": "5feb18c0-78a8-49b6-8584-75b8c1b40abc",
        "module": "CS2030",
        "title": "Watch lecture",
      }
    }
  />
  <Connect(TaskListItem)
    dashboard={true}
    key="73c8ac1a-1d06-46ee-a39e-e656e1e41db0"
    showGroup={true}
    task={
      Object {
        "completed": true,
        "description": "",
        "id": "73c8ac1a-1d06-46ee-a39e-e656e1e41db0",
        "module": "CS2040S",
        "title": "Do problem set",
      }
    }
  />
  <Connect(TaskListItem)
    dashboard={true}
    key="d2b10492-853d-4290-baf8-cb47cd4ec04e"
    showGroup={true}
    task={
      Object {
        "completed": true,
        "description": "",
        "id": "d2b10492-853d-4290-baf8-cb47cd4ec04e",
        "module": "",
        "title": "Read a book everyday",
      }
    }
  />
</div>
`;

exports[`render should render TaskList with tasks sorted by name in reverse 1`] = `
<div
  className="content-container list"
>
  <Connect(TaskListHeader) />
  <Connect(TaskListItem)
    dashboard={true}
    key="5feb18c0-78a8-49b6-8584-75b8c1b40abc"
    showGroup={true}
    task={
      Object {
        "completed": false,
        "deadline": 1591329600000,
        "description": "Catch up on lectures!",
        "id": "5feb18c0-78a8-49b6-8584-75b8c1b40abc",
        "module": "CS2030",
        "title": "Watch lecture",
      }
    }
  />
  <Connect(TaskListItem)
    dashboard={true}
    key="a1050220-fa88-4dc2-aa9d-9521ae95de47"
    showGroup={true}
    task={
      Object {
        "completed": false,
        "description": "",
        "id": "a1050220-fa88-4dc2-aa9d-9521ae95de47",
        "module": "CS2030",
        "title": "Do lab",
      }
    }
  />
  <Connect(TaskListItem)
    dashboard={true}
    key="d2b10492-853d-4290-baf8-cb47cd4ec04e"
    showGroup={true}
    task={
      Object {
        "completed": true,
        "description": "",
        "id": "d2b10492-853d-4290-baf8-cb47cd4ec04e",
        "module": "",
        "title": "Read a book everyday",
      }
    }
  />
  <Connect(TaskListItem)
    dashboard={true}
    key="73c8ac1a-1d06-46ee-a39e-e656e1e41db0"
    showGroup={true}
    task={
      Object {
        "completed": true,
        "description": "",
        "id": "73c8ac1a-1d06-46ee-a39e-e656e1e41db0",
        "module": "CS2040S",
        "title": "Do problem set",
      }
    }
  />
</div>
`;

exports[`render should render TaskList with tasks ungrouped 1`] = `
<div
  className="content-container list"
>
  <Connect(TaskListHeader) />
  <Connect(TaskListItem)
    dashboard={true}
    key="5feb18c0-78a8-49b6-8584-75b8c1b40abc"
    showGroup={true}
    task={
      Object {
        "completed": false,
        "deadline": 1591329600000,
        "description": "Catch up on lectures!",
        "id": "5feb18c0-78a8-49b6-8584-75b8c1b40abc",
        "module": "CS2030",
        "title": "Watch lecture",
      }
    }
  />
  <Connect(TaskListItem)
    dashboard={true}
    key="a1050220-fa88-4dc2-aa9d-9521ae95de47"
    showGroup={true}
    task={
      Object {
        "completed": false,
        "description": "",
        "id": "a1050220-fa88-4dc2-aa9d-9521ae95de47",
        "module": "CS2030",
        "title": "Do lab",
      }
    }
  />
  <Connect(TaskListItem)
    dashboard={true}
    key="d2b10492-853d-4290-baf8-cb47cd4ec04e"
    showGroup={true}
    task={
      Object {
        "completed": true,
        "description": "",
        "id": "d2b10492-853d-4290-baf8-cb47cd4ec04e",
        "module": "",
        "title": "Read a book everyday",
      }
    }
  />
  <Connect(TaskListItem)
    dashboard={true}
    key="73c8ac1a-1d06-46ee-a39e-e656e1e41db0"
    showGroup={true}
    task={
      Object {
        "completed": true,
        "description": "",
        "id": "73c8ac1a-1d06-46ee-a39e-e656e1e41db0",
        "module": "CS2040S",
        "title": "Do problem set",
      }
    }
  />
</div>
`;
